llvm/lib/Transforms/Vectorize/VPlanRecipes.cpp

```diff
@@ -1442,6 +1451,9 @@ InstructionCost VPWidenRecipe::computeCost(ElementCount VF,
     Type *VectorTy = toVectorTy(Ctx.Types.inferScalarType(this), VF);
     return Ctx.TTI.getArithmeticInstrCost(Instruction::Mul, VectorTy, CostKind);
   }
+  case Instruction::ExtractValue:
+    return Ctx.TTI.getInstructionCost(cast<Instruction>(getUnderlyingValue()),
```

Comment:
Hmm, this is actually asking for the cost of the scalar version of the extract. Looking at llvm/Analysis/TargetTransformInfoImpl.h::getInstructionCost it seems like we return TTI::TCC_Free anyway so I guess this is ok.

Reply:
This was just following what ``LoopVectorizationCostModel::getInstructionCost(Instruction *I, ElementCount VF)`` does (which is called by `getLegacyCost()`), which appears to ignore the VF for `extractvalue`.

Reply:
This should query the cost using the ExtractValue opcode, if possible, instead of requiring there the be an underlying instruction.